src/main/js/diff/DiffRoute.tsx

```diff
@@ -45,11 +44,22 @@ type Props = {
 };
 
 const DiffRoute: FC<Props> = ({ repository, pullRequest, source, target }) => {
-  const [diffState, dispatch] = useReducer(reducer, createInitialState(pullRequest?.markedAsReviewed || []));
-  const { error, loading, links } = useComments(pullRequest, dispatch);
   const { t } = useTranslation("plugins");
+  const { data: comments, isLoading, error } = useComments(repository, pullRequest);
+  const [diffState, setDiffState] = useState<DiffState>({
+    lines: {},
+    files: {},
+    comments: [],
+    reviewedFiles: pullRequest?.markedAsReviewed || []
+  });
 
-  const fileContentFactory: FileContentFactory = file => {
+  useEffect(() => {
+    if (comments) {
+      updateDiffStateForComments(comments, diffState, setDiffState);
+    }
+  }, [comments]);
```

Comment:
I think this dangerously, because changing the diffState with `setDiffState` is an asynchronous function. If more than one thing changes the diffState this can lead to wrong state, because the second action sees the old `diffState`. Furthermore `diffState` is a direct dependency which is used in an effect hook, it should be declared as dependency.